packages/local_auth/local_auth_darwin/darwin/local_auth_darwin/Sources/local_auth_darwin/LocalAuthPlugin.swift

```diff
@@ -0,0 +1,421 @@
+        // fire the completion.
+        completion(
+          .success(
+            AuthResultDetails(
+              result: .failure,
+              errorMessage: "evaluatePolicy failed without an error"
+            )))
+      }
+    }
+  }
+
+  func deviceCanSupportBiometrics() throws -> Bool {
+    let context = authContextFactory.createAuthContext()
+    var authError: NSError?
+    // Check if authentication with biometrics is possible.
+    if context.canEvaluatePolicy(
+      .deviceOwnerAuthenticationWithBiometrics,
+      error: &authError)
+    {
+      if authError == nil {
+        return true
+      }
+    }
+    // If not, check if it is because no biometrics are enrolled (but still present).
+    if let authError = authError {
+      if authError.code == LAError.biometryNotEnrolled.rawValue {
+        return true
+      }
+      // Biometry hardware is available, but possibly permissions were denied.
+      if authError.code == LAError.biometryNotAvailable.rawValue
+        && context.biometryType != LABiometryType.none
+      {
+        return true
+      }
+    }
+
+    return false
+  }
+
+  func getEnrolledBiometrics() throws -> [AuthBiometric] {
+    let context = authContextFactory.createAuthContext()
+    var biometrics: [AuthBiometric] = []
+    if context.canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: nil) {
+      if #available(macOS 10.15, iOS 11.0, *) {
+        if context.biometryType == LABiometryType.faceID {
+          biometrics.append(AuthBiometric.face)
+          return biometrics
+        }
+      }
+      if context.biometryType == LABiometryType.touchID {
+        biometrics.append(AuthBiometric.fingerprint)
+      }
+    }
+    return biometrics
+  }
+
+  func isDeviceSupported() throws -> Bool {
+    let context = authContextFactory.createAuthContext()
+    return context.canEvaluatePolicy(.deviceOwnerAuthentication, error: nil)
+  }
+
+  // MARK: Private Methods
+
+  @MainActor
+  private func showAlert(
+    message: String,
+    dismissButtonTitle: String,
+    openSettingsButtonTitle: String?,
+    completion: @escaping (Result<AuthResultDetails, Error>) -> Void
+  ) {
+    #if os(macOS)
+      var alert = alertFactory.createAlert()
+      alert.messageText = message
+      alert.addButton(withTitle: dismissButtonTitle)
+      if let window = viewProvider.view?.window {
+        alert.beginSheetModal(for: window) { [weak self] code in
+          self?.handleResult(succeeded: false, completion: completion)
+        }
+      } else {
+        alert.runModal()
+        self.handleResult(succeeded: false, completion: completion)
+      }
+    #elseif os(iOS)
+      // TODO(stuartmorgan): Get the view controller from the view provider once it's possible.
+      // See https://github.com/flutter/flutter/issues/104117.
+      if let controller = UIApplication.shared.delegate?.window??.rootViewController {
```

Comment:
nit: this is a good use case for `guard let`

Reply:
Converted to `guard`.